apps/guide/src/content/03-creating-your-bot/03-adding-commands.mdx

```diff
@@ -0,0 +1,146 @@
+3. The [command deployment script](command-deployment.html), to register your slash commands with Discord so they appear in the interface.
+
+These steps can be done in any order, but **all are required** before the commands are fully functional.
+
+On this page, you'll complete Step 1. Make sure to also complete the other pages linked above!
+
+</Alert>
+
+## Individual command files
+
+Create a new folder named _`commands`_, which is where you'll store all of your command files.
+
+At a minimum, the definition of a slash command must have a name and a description. Slash command names must be between 1-32 characters and contain no capital letters, spaces, or symbols other than _`-`_ and _`_`_. Using the builder, a simple _`ping`\_ command definition would look like this:
+
+<CH.Code>
+
+```js
+export const data = {
+	name: 'ping',
+	description: 'Replies with Pong!',
+};
+```
+
+</CH.Code>
+
+A slash command also requires a function to run when the command is used, to respond to the interaction. Using an interaction response method confirms to Discord that your bot successfully received the interaction, and has responded to the user. Discord enforces this to ensure that all slash commands provide a good user experience (UX). Failing to respond will cause Discord to show that the command failed, even if your bot is performing other actions as a result.
+
+The simplest way to acknowledge and respond to an interaction is the _`interaction.reply()`_ method. Other methods of replying are covered on the [Response methods](/slash-commands/response-methods.md) page later in this section.
+
+<CH.Code>
+
+```js
+export async function execute(interaction) {
+	await interaction.reply('Pong!');
+}
+```
+
+</CH.Code>
+
+Put these two together by creating a `commands/ping.js` file for your first command. Inside this file, you're going to define and export two items.
+
+- The `data` property, which will provide the command definition shown above for registering to Discord.
+- The `execute` method, which will contain the functionality to run from our event handler when the command is used.
+
+The _`export`_ keyword ensures these values can be imported and read by other files; namely the command loader and command deployment scripts mentioned earlier.
+
+<CH.Code>
+
+```js commands/ping.js
+export const data = {
+	name: 'ping',
+	description: 'Replies with Pong!',
+};
+
+export async function execute(interaction) {
+	await interaction.reply('Pong!');
+}
+```
+
+</CH.Code>
+
+<Alert title="Tip" type="success">
+[`module.exports`](https://nodejs.org/api/modules.html#modules_module_exports) is how you export data in Node.js so that you can [`require()`](https://nodejs.org/api/modules.html#modules_require_id) it in other files.
+
+If you need to access your client instance from inside a command file, you can access it via `interaction.client`. If you need to access external files, packages, etc., you should `require()` them at the top of the file.
+
+</Alert>
```

Comment:
```suggestion
<Alert title="Tip" type="success">
	[`module.exports`](https://nodejs.org/api/modules.html#modules_module_exports) is how you export data in Node.js so that you can [`require()`](https://nodejs.org/api/modules.html#modules_require_id) it in other files.

	If you need to access your client instance from inside a command file, you can access it via `interaction.client`. If you need to access external files, packages, etc., you should `require()` them at the top of the file.
</Alert>
```